pulsar-transaction/buffer/src/main/java/org/apache/pulsar/transaction/buffer/impl/PersistentTransactionBufferProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.apache.pulsar.transaction.buffer.impl;
+
+import java.util.concurrent.CompletableFuture;
+import lombok.Setter;
+import org.apache.bookkeeper.mledger.AsyncCallbacks;
+import org.apache.bookkeeper.mledger.ManagedLedger;
+import org.apache.bookkeeper.mledger.ManagedLedgerException;
+import org.apache.pulsar.broker.service.BrokerService;
+import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.transaction.buffer.TransactionBuffer;
+import org.apache.pulsar.transaction.buffer.TransactionBufferProvider;
+
+public class PersistentTransactionBufferProvider implements TransactionBufferProvider {
+
+    private final BrokerService brokerService;
+    private final String topic;
+    private final String txnTopic;
+
+    public PersistentTransactionBufferProvider(BrokerService service, String topic) {
+        this.brokerService = service;
+        // TODO: get the transaction topic name by the TopicName.getPersistentNamingEncoding(isTxn)
+        this.txnTopic = TopicName.get(topic).getPersistenceNamingEncoding() + "/_txnlog";
+        this.topic = topic;
+    }
+
+    @Override
+    public CompletableFuture<TransactionBuffer> newTransactionBuffer() {
+        CompletableFuture<TransactionBuffer> newBufferFuture = new CompletableFuture<>();
+        brokerService.getManagedLedgerConfig(TopicName.get(topic)).whenComplete((config, err) -> {
```

Comment:
why not:

```
return brokerService.getManagerLedgerConfig(...)
    .thenCompose(brokerService.getManagerLedgerFactory().asyncOpen(...));
```

Reply:
The config can change before opening the ledger. We need to set the `createIfMissing` as true.

Reply:
why not just add an overload method `getManagedLedgerConfig(TopicName.get(topic), createIfMissing)`? 

It is better to put all the configurations in one central place rather than scattering the setters all overall the code.